relay-general/src/pii/minidumps.rs

```diff
@@ -542,4 +592,133 @@ mod tests {
+        }
+    }
+
+    #[test]
+    fn test_stack_scrubbing_valuetype_selector() {
+        let scrubber = TestScrubber::new(
+            "linux.dmp",
+            include_bytes!("../../../tests/fixtures/linux.dmp"),
+            serde_json::json!(
+                {
+                    "applications": {
+                        "$minidump.$binary": ["@anything:mask"],
+                    }
+                }
+            ),
+        );
+        for stack in scrubber.stacks(Which::Scrubbed) {
+            assert!(stack.iter().all(|b| *b == '*' as u8));
+        }
+    }
+
+    #[test]
+    fn test_stack_scrubbing_valuetype_not_fully_qualified() {
+        // Not fully qualified valuetype should not touch the stack: FAILS
+        let scrubber = TestScrubber::new(
+            "linux.dmp",
+            include_bytes!("../../../tests/fixtures/linux.dmp"),
+            serde_json::json!(
+                {
+                    "applications": {
+                        "$binary": ["@anything:mask"],
+                    }
+                }
+            ),
+        );
+        for stack in scrubber.stacks(Which::Scrubbed) {
+            assert!(!stack.iter().all(|b| *b == '*' as u8));
+        }
+    }
+
+    #[test]
+    fn test_stack_scrubbing_wildcard() {
+        // Wildcard should not touch the stack: FAILS
+        let scrubber = TestScrubber::new(
+            "linux.dmp",
+            include_bytes!("../../../tests/fixtures/linux.dmp"),
+            serde_json::json!(
+                {
+                    "applications": {
+                        "$minidump:*": ["@anything:mask"],
+                    }
+                }
+            ),
+        );
+        for stack in scrubber.stacks(Which::Scrubbed) {
+            assert!(!stack.iter().all(|b| *b == '*' as u8));
+        }
+    }
+
+    #[test]
+    fn test_stack_scrubbing_deep_wildcard() {
+        // Wildcard should not touch the stack: FAILS
+        let scrubber = TestScrubber::new(
+            "linux.dmp",
+            include_bytes!("../../../tests/fixtures/linux.dmp"),
+            serde_json::json!(
+                {
+                    "applications": {
+                        "$attachments:**": ["@anything:mask"],
+                    }
+                }
+            ),
+        );
+        for stack in scrubber.stacks(Which::Scrubbed) {
+            assert!(!stack.iter().all(|b| *b == '*' as u8));
+        }
+    }
+
+    #[test]
+    fn test_stack_scrubbing_binary_not_stack() {
+        let scrubber = TestScrubber::new(
+            "linux.dmp",
+            include_bytes!("../../../tests/fixtures/linux.dmp"),
+            serde_json::json!(
+                {
+                    "applications": {
+                        "$binary && !stack_memory": ["@anything:mask"],
+                    }
+                }
+            ),
+        );
+        for stack in scrubber.stacks(Which::Scrubbed) {
+            assert!(!stack.iter().all(|b| *b == '*' as u8));
```

Comment:
assert not (`!`) all, does that assert that the stack got partially scrubbed or not at all?

same for other assertions